yarn-project/simulator/src/public/abstract_phase_manager.ts

```diff
@@ -208,40 +238,32 @@ export abstract class AbstractPhaseManager {
     return calls;
   }
 
-  // REFACTOR: Do not return an array and instead return a struct with similar shape to that returned by `handle`
   protected async processEnqueuedPublicCalls(
     tx: Tx,
     previousPublicKernelOutput: PublicKernelCircuitPublicInputs,
-  ): Promise<
-    [
-      PublicKernelCircuitPrivateInputs[],
-      PublicKernelCircuitPublicInputs,
-      UnencryptedFunctionL2Logs[],
-      SimulationError | undefined,
-      NestedProcessReturnValues[],
-      Gas,
-    ]
-  > {
-    let kernelOutput: PublicKernelCircuitPublicInputs = previousPublicKernelOutput;
-    const publicKernelInputs: PublicKernelCircuitPrivateInputs[] = [];
-
+  ): Promise<TxPublicCallsResult> {
     const enqueuedCalls = this.extractEnqueuedPublicCalls(tx);
 
     if (!enqueuedCalls || !enqueuedCalls.length) {
-      return [[], kernelOutput, [], undefined, [], Gas.empty()];
+      return {
+        publicProvingInformation: [],
+        kernelOutput: previousPublicKernelOutput,
+        newUnencryptedLogs: [],
+        returnValues: [],
+        gasUsed: Gas.empty(),
+      };
     }
 
-    const newUnencryptedFunctionLogs: UnencryptedFunctionL2Logs[] = [];
-
-    // Transaction fee is zero for all phases except teardown
-    const transactionFee = this.getTransactionFee(tx, previousPublicKernelOutput);
-
     // TODO(#1684): Should multiple separately enqueued public calls be treated as
     // separate public callstacks to be proven by separate public kernel sequences
     // and submitted separately to the base rollup?
 
+    const provingInformationList: PublicProvingInformation[] = [];
+    const newUnencryptedFunctionLogs: UnencryptedFunctionL2Logs[] = [];
+    // Transaction fee is zero for all phases except teardown
+    const transactionFee = this.getTransactionFee(tx, previousPublicKernelOutput);
     let gasUsed = Gas.empty();
-
+    let kernelPublicOutput: PublicKernelCircuitPublicInputs = previousPublicKernelOutput;
```

Comment:
We pretty much renamed `kernelOutput` to `kernelPublicOutput` here. Should it be `publicKernelOutput`? Otherwise I'm not sure the word public is adding any value here.

Reply:
Yeah, I struggled to make sense of this. The original code (well, and this one) actually handle 2 types of outputs/inputs for the kernel:
1. the PublicKernelPrivateInputs (which are put into the private proving info); and
2. the PublicKernelPublicInputs (which are used/returned as single kernelOutputs)

the idea to use public here was to make it evident that these were the PUBLIC kernel PUBLIC inputs/outputs (and not the private ones).